packages/bot-skeleton/src/scratch/blocks/Math/math_on_list.js

```diff
@@ -50,8 +50,8 @@ Blockly.Blocks.math_on_list = {
     },
     getRequiredValueInputs() {
         return {
-            LIST: input => {
-                return input !== 'list';
+            LIST: () => {
+                return this.childBlocks_[0]?.category_ !== 'variables' && this.childBlocks_[0]?.category_ !== null;
```

Comment:
this return condition always returns `false`
you may need `or` instead of `and` here
Also, we have another situation that may cause an issue, it's possible if a client creates a custom_function to return a `list` in this situation we get the error again
you should check if the input `type` is `procedures_callreturn` and then get the `RETURN` value and check if it's `variables` or not